scipy/stats/_distn_infrastructure.py

```diff
@@ -2230,6 +2279,33 @@ def _nnlf_and_penalty(self, x, args):
             return -np.sum(logpdf[finite_logpdf], axis=0) + penalty
         return -np.sum(logpdf, axis=0)
 
+    def _censored_nnlf_and_penalty(self, x, args):
+        # Modified version of _nnlf_and_penalty.
+        cond0 = ~self._support_mask(x, *args)
+        n_bad = np.count_nonzero(cond0)
+        if n_bad > 0:
+            x = CensoredData(x._lower[~cond0], x._upper[~cond0])
+
+        # logpdf of the noncensored data.
+        logpdf_nc = self._logpdf(x._lower[x._not_censored], *args)
+        # logcdf of the left-censored data.
+        logcdf_lc = self._logcdf(x._upper[x._left_censored], *args)
+        # logsf of the right-censored data.
+        logsf_rc = self._logsf(x._lower[x._right_censored], *args)
+        # Probability of the interval-censored data.
+        # TO DO: Use a smarter "delta CDF" function for this.
```

Comment:
Good idea. Let's add a private method in `rv_continuous`.

Reply:
Done.  The private method is `_delta_cdf`.